src/features/domain-settings/hooks/useDomainSettingsData.ts

```diff
@@ -0,0 +1,45 @@
+import {
+	useWeb3,
+	useDomainData,
+	useDomainMetadata,
+	useIsDomainMetadataLocked,
+} from '../../../lib/hooks';
+import { getDomainId } from '../../../lib/util';
+import { truncateAddress } from '@zero-tech/zui/utils';
+
+export const useDomainSettingsData = (zna: string) => {
+	const domainId = getDomainId(zna);
+
+	const { account } = useWeb3();
+
+	const { data: domain, isLoading: isLoadingDomainData } =
+		useDomainData(domainId);
+
+	const { data: metadata, isLoading: isLoadingMetadata } =
+		useDomainMetadata(domainId);
+
+	const { data: metadataLockedStatus, isLoading: isLoadingLockedStatus } =
```

Comment:
Are there multiple `metadataLockedStatus` values, or is it just `true`, `false`, or `undefined`? If it's the latter, can we change this to `isMetadataLocked`? I generally like to start all boolean variable names with `is` so it's obvious what values they can contain!